packages/markdown/remark/src/rehype-collect-headings.ts

```diff
@@ -31,7 +34,18 @@ export function rehypeHeadingIds(): ReturnType<RehypePlugin> {
 				}
 				if (rawNodeTypes.has(child.type)) {
 					if (isMDX || codeTagNames.has(parent.tagName)) {
-						text += child.value;
+						let value = child.value;
+						const astroData = safelyGetAstroData(file.data);
```

Comment:
Good catch! Nit: might move this call to the top of the function to avoid reparsing on every pass